### ISSUE TITLE
Public: Use ARK in URLs

### ISSUE THREAD

**Issue by andrewbenedictwallace** [UCLALibrary/ursus#170]
Currently URLs use a fedora ID that changes each time we ingest.

Acceptance criteria:
- [ ] Single-page urls use UCLA ARK identifiers

depends on github.com/UCLALibrary/californica/issues/358
child of github.com/UCLALibrary/californica/issues/336

**Comment by val99erie** [comment]
I'd be curious to know if anyone else in the Hyrax community has tried changing solr's unique key to a different field.

I worry that some of the underlying gems in the Hyrax ecosystem won't behave properly if the unique key is changed to a different field.  Some classes seem to have baked-in assumptions that the unique key field is 'id'.

I tried changing the uniqueKey in my development workspace to see how hard it would be.  It's not trivially easy.

I think we would have to change the indexing for every model in the Hyrax stack to make sure that the required 'ark' field gets added to every solr document.  I'm not sure if there is a clean way to do that without just monkey-patching the indexer from the active-fedora gem.  I tried monkey-patching the indexer in my development workspace, and I was able to successfully save records.

But, then I ran into another problem when I tried to load the show page for a Work record.  The query failed to find the correct record because it's searching the 'id' field instead of the 'ark' field.  At this point I think we need to make changes to the request handlers in the solr config and/or more changes to the underlying gem behavior.

Another difficulty is that if we make changes to the solr schema, it might mean re-building the production environment.  (It's not impossible, I'm just pointing out that it would take some work.)

Another thing to consider is how far off the path this would take us from vanilla Hyrax behavior.  We might be making it very difficult for us to upgrade in the future.

I spent a couple hours travelling down that path in my development environment, and it seems like there are lots of little trails we'd need to follow to get everything working.

I recommend that we try another approach first.  Making code changes without altering the solr config seems less risky to me.
